### ISSUE TITLE
Dashboard Component: Create server components for dashboard

### ISSUE THREAD

**Issue by Cooryd** [elifesciences/elife-xpub#1714]
## Background

__Complete #1711 before starting__

__This issue relates to both the movement of code AND refactoring, it will surely need breaking down into even smaller tasks!__

__This does not need to be imported and used within the app itself yet, that will be a separate issue__

As part of #1705 we need to move the server side Dashboard related code into our new `component-dashboard` directory. This should also aim to split the `use-case` (controller) and `resolver` code out to make the code easier to modify and test. Resolvers should be injecting models and clients into the use-case functions so that unit test around the use-cases can be ignorant of the models implementation and pass down mocked interfaces instead.

## Task

__See [Hindawi's component-dashboard package](https://gitlab.com/hindawi/xpub/xpub-review/tree/develop/packages/component-dashboard/server) for more context on what we're aiming for__

directory/ file structure should look similar to this:

```
/server
  /use-cases
  resolvers.js
  typeDefs.grapqls
```

### use-cases

This should have an `index.js` that pulls in each individual use-case function and exports them as a single object for use by the resolver.

each use-case will have a single file which will be set up in the following pattern so that it may take injected dependencies:

```js

const initialize = ({
 ... dependencies
}) => ({
  execute: async ({ ...parameters }) => {
   //use-case code here
  }
})

```

### resolvers.js

This should be very lightweight as all functionality is taken care of in the use-cases. This file should simply import in the model and provider dependencies (ie: `logger` and `manuscript-model`) as well as the use-cases object then call them by passing the correct dependencies and params

example: 

```javascript
const Manuscript = require('@elifesciences/xpub-model/Manuscript')
const useCases = require('./use-cases')

const resolvers = {
  Query: {
    async getManuscripts(_, { input }, ctx) {
      return useCases.getManuscripts
        .initialize({ Manuscript })
        .execute({ input, userId: ctx.user })
    },
  },
```

**Comment by diversemix** [comment]
Having talked with @Cooryd over lunch - I think we need to keep the refactoring to a minimum in the first round of changes ... As then the refactoring work can happen in isolation from other components :)

**Comment by Cooryd** [comment]
This should be completed in two phases then 
1. Move the code into the `packages/component-dashboard/server` directory in a structure similar to that described above without refactoring too much. (this work can relate to this issue)
2. Refactor the code to make use of dependency injected models into the `use-cases` and re-write tests to make use of the injected dependencies. (will need to create another issue for this step)

**Comment by Cooryd** [comment]
Phase 1 of this refactoring work will create the component-xxxx directories and move the code across without making any major code changes.
